juju/juju
# 2.4 Fix for Ubuntu Hosts With LXD Not Installed

## Description of change

Addresses an issue specifically with Ubuntu host machines *without* LXD installed, where the provisioner set-up for container infrastructure causes an attempted connection to the LXD server before the installation/config has been assured. 

Also checks whether LXD is running before attempting to update its proxy configuration. If it is not, then that fact is logged and a nil error is returned.

Includes some drive-by spelling fixes.

## QA steps

- Bootstrap to local LXD.
- Add a new machine.
- Exec `sudo apt remove lxd` in the new container.
- `juju add-machine 0:lxd`
- Check machine-0 log for errors.

## Documentation changes

None.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1780388
